### ISSUE TITLE
Sudden influx of ILM cold shards turn cluster RED before it is autoscaled

### ISSUE THREAD

**Issue by kingherc** [elastic/elasticsearch#97370]
### Elasticsearch Version

8.8.1

### Java Version

_bundled_

### OS Version

cloud

### Problem Description

The autoscaler reactive decider takes into account unassigned and unmovable shards. It's OK for ILM moves. However, cold shards are directly created on cold nodes, since they are mounted from snapshots.

### Steps to Reproduce

The information on the soon-to-be-created shards from ILM is missed in the decider. This means that a sudden influx of cold shards on cold nodes that are nearing their disk limits, can tip them over and make the cluster RED. Ideally, we would like a better co-operation between the ILM cold phase and the decider. Maybe something akin to the [WaitForDataTierStep](https://github.com/elastic/elasticsearch/pull/73435) but that takes into account all possible concurrent ILM cold phases.

**Comment by kingherc** [comment]
After the team discussion, we determined that ES is turning RED because the cold shards are considered unassigned primaries. We discussed a possible mechanism that avoids creating the cold shards until the disk space becomes available, but we decided against this because any such check would not be guaranteed. The unassigned cold shards are an important part of the autoscaling mechanism, since they are the signal which indicates that a scale-up is needed. Thus, the RED health status should not be construed as alarming, since the data is still safely in the respective hot shards that are becoming cold, and will remain there until the cold shards are assigned so that searches can still search them. That is why the ES autoscale still correctly tells to scale – which would then make the cluster green again. 

In conclusion, RED cluster health is not useful as a direct indicator of service unavailability, and it is not appropriate to build alerts using this indicator. Instead, use the [Health Report API](https://www.elastic.co/guide/en/elasticsearch/reference/8.8/health-api.html). However, it's our understanding that the Health Report API would today also make the same mistake. We have opened [#97606](https://github.com/elastic/elasticsearch/issues/97606) to report that problem, and will close this issue.
